cijoe/scripts/latency_plotter.py

```diff
@@ -27,278 +27,214 @@
-
     with path.open() as jfd:
         data = json.load(jfd)
 
-    plot_attributes = plot_attributes_from_step(step)
+    plot_attributes = get_plot_attributes(step)
 
-    x = "iodepth"
     y = "lat"
 
-    data = data_as_a_function_of(data, x, y)
-
-    grouped = build_plot_groups(data)
-
-    for group, dset in grouped.items():
-        plt.clf()
+    max_y = max(map(lambda item: item[1][y], data))
+    all_groups = set(map(lambda item: item[1]["ctx"]["group"], data))
+
+    # Create a plot for each group, showing the scalability of the xNVMe overhead
+    for group in all_groups:
+        dset = data_as_a_function_of(
+            data,
+            x="iodepth",
+            y=y,
+            filter=lambda item: item["ctx"]["group"] == group
+            and item["ctx"]["iosize"] == 4096,
+        )
+        draw_bar_plot(
+            dset, plot_attributes, xlabel="iodepth", y_limit=max_y * PLOT_SCALE
+        )
 
-        for _, samples in dset.items():
-            label = samples["label"]
-            attrs = plot_attributes["legends"].get(label, None)
-            if attrs is None:
-                log.error(f"Missing plot-attributes for label({label})")
-                continue
+        os.makedirs(artifacts, exist_ok=True)
+        plt.savefig(artifacts / f"fio_barplot_{group}_iodepth.png")
 
-            xs, ys = [list(t) for t in zip(*sorted(zip(samples["xs"], samples["ys"])))]
-            plt.plot(
-                xs,
-                ys,
-                label=attrs["legend"],
-                color=attrs["color"],
-                marker=attrs["marker"],
-            )
+    for group in all_groups:
+        dset = data_as_a_function_of(
+            data,
+            x="iosize",
+            y=y,
+            filter=lambda item: item["ctx"]["group"] == group
+            and item["ctx"]["iodepth"] == 1,
+        )
+        draw_bar_plot(
+            dset, plot_attributes, xlabel="iosize", y_limit=max_y * PLOT_SCALE
+        )
 
-        plt.xlabel(x)
-        plt.ylabel(y)
+        os.makedirs(artifacts, exist_ok=True)
+        plt.savefig(artifacts / f"fio_barplot_{group}_iosize.png")
 
-        plt.legend(
-            bbox_to_anchor=(0, 0.95, 1, 0.2),
-            loc="lower center",
-            ncol=2,
-            facecolor="white",
-            framealpha=1,
-        )
-        plt.ylim(plot_attributes["limits"]["lineplot"]["y_lim"])
+    # Create a plot, showing the overhead of xNVMe at queue depth = 1
+    dset = data_as_a_function_of(
+        data,
+        x="iodepth",
+        y=y,
+        filter=lambda item: item["ctx"]["group"] != "null"
+        and item["ctx"]["iodepth"] == 1
+        and item["ctx"]["iosize"] == 4096,
+    )
+    draw_bar_plot(dset, plot_attributes)
 
-        os.makedirs(args.output / "artifacts", exist_ok=True)
-        plt.savefig(args.output / "artifacts" / f"fio_lineplot_{group}.png")
+    os.makedirs(artifacts, exist_ok=True)
+    plt.savefig(artifacts / "barplot_qd1.png")
```

Comment:
Perhaps this plot should also have the "fio_` prefix like the others, in case we end up having more barplots with qd1